kclvm/sema/src/core/symbol.rs

```diff
@@ -0,0 +1,389 @@
+                owner_name + "." + &type_symbol.name
+            }
+            SymbolKind::Rule => {
+                let rule_symbol = self.rules.get(symbol_ref.get_id()).unwrap();
+                let owner_name = self.get_fully_qualified_name(rule_symbol.owner);
+                owner_name + "." + &rule_symbol.name
+            }
+            SymbolKind::Package => self.packages.get(symbol_ref.get_id()).unwrap().name.clone(),
+            SymbolKind::Function => todo!(),
+            SymbolKind::Value => todo!(),
+            SymbolKind::Unresolved => todo!(),
+            SymbolKind::Dummy => todo!(),
+        }
+    }
+
+    pub fn build_fully_qualified_name_map(&mut self) {
+        for (id, _) in self.packages.iter() {
+            let symbol_ref = SymbolRef {
+                id,
+                kind: SymbolKind::Package,
+            };
+            self.fully_qualified_name_map
+                .insert(self.get_fully_qualified_name(symbol_ref), symbol_ref);
+        }
+
+        for (id, _) in self.schemas.iter() {
+            let symbol_ref = SymbolRef {
+                id,
+                kind: SymbolKind::Schema,
+            };
+            self.fully_qualified_name_map
+                .insert(self.get_fully_qualified_name(symbol_ref), symbol_ref);
+        }
+
+        for (id, _) in self.type_aliases.iter() {
+            let symbol_ref = SymbolRef {
+                id,
+                kind: SymbolKind::TypeAlias,
+            };
+            self.fully_qualified_name_map
+                .insert(self.get_fully_qualified_name(symbol_ref), symbol_ref);
+        }
+
+        for (id, _) in self.attributes.iter() {
+            let symbol_ref = SymbolRef {
+                id,
+                kind: SymbolKind::Attribute,
+            };
+            self.fully_qualified_name_map
+                .insert(self.get_fully_qualified_name(symbol_ref), symbol_ref);
+        }
+
+        for (id, _) in self.rules.iter() {
+            let symbol_ref = SymbolRef {
+                id,
+                kind: SymbolKind::Rule,
+            };
+            self.fully_qualified_name_map
+                .insert(self.get_fully_qualified_name(symbol_ref), symbol_ref);
+        }
+    }
+
+    fn resolve_symbol(&self, symbol_ref: SymbolRef, package_info: &PackageInfo) -> SymbolRef {
+        if matches!(symbol_ref.get_kind(), SymbolKind::Unresolved) {
+            let unresolved_symbol = self.unresolved.get(symbol_ref.get_id()).unwrap();
+            let target_symbol = self
+                .fully_qualified_name_map
+                .get(&unresolved_symbol.get_fully_qualified_name(package_info));
+            match target_symbol {
+                Some(target_symbol) => *target_symbol,
+                None => symbol_ref,
+            }
+        } else {
+            symbol_ref
+        }
+    }
+
+    pub fn replace_unresolved_symbol(&mut self, package_db: &PackageDB) {
```

Comment:
I noticed that this function only handles schema symbols, is this expected? Do other symbols such as type alias symbols need to be processed?